config/datocms/migrations/1752588128_featSpecialPagesContent.ts

```diff
@@ -0,0 +1,196 @@
+        fieldValue = rest;
+      }
+      for (const key in fieldValue) {
+        fieldValue[key] = processField(fieldValue[key]);
+      }
+      return fieldValue;
+    }
+    return field;
+  }
+
+  const [home_page, not_found_page] = await Promise.all([
+    { record: lastPublishedHomePageRecord, slug: 'home' },
+    { record: lastPublishedNotFoundPageRecord, slug: 'not-found' },
+  ].map(async ({ record, slug }) => {
+    const { id, type, item_type, meta, creator, ...rest } = record;
+    console.log(`Migrating ${type} to Page record with slug "${slug}"`);
+    const localizedSlug = locales
+      .reduce((acc, locale) => ({
+        ...acc,
+        [locale]: slug,
+      }), {} as Record<typeof locales[number], string>);
+
+    // Insert the new Page record
+    return client.items.create({
+      ...processContent(rest),
+      creator,
+      item_type: { type: 'item_type', id: pageModel.id },
+      slug: locales.length > 1
+        ? localizedSlug
+        : localizedSlug[locales[0]],
+    })
+      // Publish the new Page record
+      .then(async ({ id }) => (await client.items.publish(id)))
+      // Update the new Page with original's most recent unpublished version
+      .then(async (page) => {
+        const lastSavedSingletonRecord = await client.items.find(record.id, { nested: true, version: 'current' });
+        const { id, type, item_type, meta, creator, ...rest } = lastSavedSingletonRecord;
+        if (meta.status !== 'published') {
+          client.items.update(page.id, processContent(rest));
+        }
+        return page;
+      })
+      // return the id of the new Page record
+      .then(({ id }) => id);
+  }));
+
+  console.log(
+    'Insert migrated pages into "\uD83D\uDDA5\uFE0F Website" (`app`)'
+  );
+  const { id, meta: { status } } = await client.items.find(appSettingsModel.singleton_item?.id);
+  const isPreMigrationAppSettingsPublished = status === 'published';
+  await client.items.update(appSettingsModel.singleton_item?.id, {
+    home_page,
+    not_found_page,
+  });
+  if (isPreMigrationAppSettingsPublished) {
+    console.log('Publish updated "\uD83D\uDDA5\uFE0F Website" (`app`)');
+    await client.items.publish(id);
+  }
+
+  if (!homePageModel.singleton_item?.id) {
+    throw new Error('Home page model does not have a singleton item');
+  }
+
+  console.log(
+    'Updating records linking to "\uD83C\uDFE0 Home" (`home_page`)'
+  );
+  const recordsReferencingHomePage = await client.items.references(
+    homePageModel.singleton_item.id,
+    { nested: true },
+  );
+
+  for (const record of recordsReferencingHomePage) {
+    const { id, type, item_type, meta, creator, ...content } = record;
+    const title = isLocalizedField(content.title)
+      ? content.title[locales[0]]
+      : content.title;
+    console.log(`Updating record "${title || id}"`);
+    await client.items.update(
+      record.id,
+      replaceReferences(content, homePageModel.singleton_item.id, home_page),
+    );
+  }
```

Comment:
this migrates any record that has a reference to the old home page to the new home page